lib/discourse_slack/slack.rb

```diff
@@ -216,5 +216,51 @@ def self.notify(id)
 
       responses
     end
+
+    def self.channels
+      return [] if SiteSetting.slack_access_token.empty?
+
+      @channels = Rails.cache.fetch("slack_channels", expires_in: 15.minutes) do
+        http = Net::HTTP.new("slack.com" , 443)
+        http.use_ssl = true
+
+        uri = URI("https://slack.com/api/channels.list" +
+          "?token=#{ SiteSetting.slack_access_token }"
```

Comment:
In Ruby, we prefer to interpolate the string instead of `+`ing them because `+`ing creates a whole new string object.